packages/components/src/progressive-disclosure-panel/README.md

```diff
@@ -0,0 +1,117 @@
+# Progressive Disclosure Panel
+
+These panels provide progressive discovery options for their children. For
+example the controls provided via block supports.
+
+## Development guidelines
+
+The `ProgressiveDisclosurePanel` creates a container with a header including a
+dropdown menu. The menu is generated automatically from the panel's children.
+Each menu item allows for the display of the corresponding child to be
+toggled on or off. The control's `onSelect` and `onDeselect` callbacks are fired
+allowing for greater control over the child e.g. resetting block attributes when
+a block support control is toggled off.
+
+Whether a child control is initially displayed or not is dependent upon
+if there has previously been a value set or the child has been flagged as
+displaying by default through the `isShownByDefault` prop. Determining whether a
+child has a value is done via the `hasValue` function provided through the
+child's props.
+
+### Usage
+
+```jsx
+import { __experimentalProgressiveDisclosurePanel as ProgressiveDisclosurePanel } from '@wordpress/components';
+import {
+	PaddingEdit,
+	hasPaddingValue,
+	resetPadding,
+	useIsPaddingDisabled,
+} from './padding';
+
+
+export function DimensionPanel( props ) {
+	const isPaddingDisabled = useIsPaddingDisabled( props );
+
+	const resetAll = () => {
+		// Reset attributes for all block support features in this panel.
+	};
+
+	return (
+		<ProgressiveDisclosurePanel
+			label={ __( 'Dimensions options' ) }
+			title={ __( 'Dimensions' ) }
+			resetAll={ resetAll }
+		>
+			{ ! isPaddingDisabled && (
+				<PaddingEdit
+					{ ...props }
+					hasValue={ hasPaddingValue }
+					label={ __( 'Padding' ) }
+					onDeselect={ resetPadding }
+				/>
+			) }
+		</ProgressiveDisclosurePanel>
+	);
+}
+```
+
+### Props
+
+#### label
+
+The label for the panel's dropdown menu.
+
+#### resetAll
+
+A function to call when the `Reset all` menu option is selected. This is passed
+through to the panel's title component.
+
+#### title
+
+Title to be displayed within the panel's title.
+
+### Sub-Components
+
+#### ProgressiveDisclosurePanelTitle
```

Comment:
There is not need to expose this component, right? If that's the case, I guess we should not expose and specify that in the documentation that the component is not exposed or even remove the docs from this file and document directly in the code of the component.

Reply:
I was following what was in the documentation for the original `Panel` here.  I can certainly remove the documentation around the panel title. I should update it with an explanation of the new panel item component as well.

Reply:
I've updated the readme to only detail the new panel item component as consumers will need to wrap their controls within that.